web-client/src/lib.rs

```diff
@@ -15,101 +16,211 @@ pub use nimiq::{
 use nimiq_network_libp2p::Multiaddr;
 
-async fn light_client() {
-    let log_settings = LogSettings {
-        level: Some(LevelFilter::DEBUG),
-        ..Default::default()
-    };
-
-    // Initialize logging with config values.
-    initialize_web_logging(Some(&log_settings)).expect("Web logging initialization failed");
-
-    // Initialize panic hook.
-    initialize_panic_reporting();
-
-    // Create config builder.
-    let mut builder = ClientConfig::builder();
-
-    // Finalize config.
-    let mut config = builder
-        .volatile()
-        .light()
-        .build()
-        .expect("Build configuration failed");
-
-    let seed = Seed {
-        address: Multiaddr::from_str("/dns4/seed1.v2.nimiq-testnet.com/tcp/8443/ws").unwrap(),
-    };
-    config.network.seeds = vec![seed];
-
-    log::debug!("Final configuration: {:#?}", config);
-
-    // Create client from config.
-    log::info!("Initializing light client");
-    let mut client: Client = Client::from_config(
-        config,
-        Box::new(|fut| {
-            spawn_local(fut);
-        }),
-    )
-    .await
-    .expect("Client initialization failed ");
-    log::info!("Web client initialized");
-
-    // Start consensus.
-    let consensus = client.take_consensus().unwrap();
-
-    log::info!("Spawning consensus");
-    spawn_local(consensus);
-
-    let consensus = client.consensus_proxy();
-
-    let zkp_component = client.take_zkp_component().unwrap();
-    spawn_local(zkp_component);
-
-    // Create the "monitor" future which never completes to keep the client alive.
-    // This closure is executed after the client has been initialized.
-    // TODO Get rid of this. Make the Client a future/stream instead.
-    let mut statistics_interval = LogSettings::default_statistics_interval();
-    let mut show_statistics = true;
-    if statistics_interval == 0 {
-        statistics_interval = 10;
-        show_statistics = false;
+#[wasm_bindgen]
+pub struct WebClient {
+    #[wasm_bindgen(skip)]
+    pub inner: Client,
+}
+
+#[wasm_bindgen]
+impl WebClient {
+    pub async fn launch() -> WebClient {
+        let log_settings = LogSettings {
+            level: Some(LevelFilter::DEBUG),
+            ..Default::default()
+        };
+
+        // Initialize logging with config values.
+        initialize_web_logging(Some(&log_settings)).expect("Web logging initialization failed");
+
+        // Initialize panic hook.
+        initialize_panic_reporting();
+
+        // Create config builder.
+        let mut builder = ClientConfig::builder();
+
+        // Finalize config.
+        let mut config = builder
+            .volatile()
+            .light()
+            .build()
+            .expect("Build configuration failed");
+
+        let seed = Seed {
+            address: Multiaddr::from_str("/dns4/seed1.v2.nimiq-testnet.com/tcp/8443/ws").unwrap(),
+        };
+        config.network.seeds = vec![seed];
+
+        log::debug!("Final configuration: {:#?}", config);
```

Comment:
We should change this to `tracing style`:
```suggestion
        log::debug!(?config, "Final configuration");
```